playbook-website/app/controllers/pages_controller.rb

```diff
@@ -67,6 +67,35 @@ def kit_show_swift
     render template: "pages/kit_show", layout: "layouts/kits"
   end
 
+  def kit_collection_show_rails
+    @kits = params[:names].split("%26")
+    @kits_array = @kits.first.split("&")
+    params[:name] ||= @kits_array[0]
+    @selected_kit = params[:name]
+    @type = "rails"
+    render template: "pages/kit_collection", layout: "layouts/fullscreen"
+  end
+
+  def kit_collection_show_react
+    @kits = params[:names].split("%26")
+    @kits_array = @kits.first.split("&")
+    params[:name] ||= @kits_array[0]
+    @selected_kit = params[:name]
+    @type = "react"
+
+    render template: "pages/kit_collection", layout: "layouts/fullscreen"
+  end
+
+  def kit_collection_show
+    @kits = params[:names].split("%26")
+    @kits_array = @kits.first.split("&")
+    params[:name] ||= @kits_array[0]
+    @selected_kit = params[:name]
+    @type = params[:type]
+
+    render template: "pages/kit_collection", layout: "layouts/fullscreen"
+  end
```

Comment:
To avoid duplication we extract some of the logic here into a private method below line 173 of your code. Let's call it `set_kit_collection`. Here's how i would do it. Mind giving this a try? 

```
def kit_collection_show_rails
  set_kit_collection("rails")
end

def kit_collection_show_react
  set_kit_collection("react")
end

def kit_collection_show
  set_kit_collection(params[:type])
end

private

def set_kit_collection(type)
  @kits = params[:names].split("%26")
  @kits_array = @kits.first.split("&")
  params[:name] ||= @kits_array[0]
  @selected_kit = params[:name]
  @type = type

  render template: "pages/kit_collection", layout: "layouts/fullscreen"
end

```

Reply:
@jasperfurniss Great idea!!! Thank you for that example, i will throw this into the my local and test it out!!

Reply:
![Screen Shot 2023-06-14 at 3 48 28 PM](https://github.com/powerhome/playbook/assets/92755007/652a6a9b-17ef-4482-a1eb-becf85d4ce98)

Rubocop doesnt like me using `set_` as a prefix. I decided to use `handle_` let me know if i should change it to something else @jasperfurniss

Reply:
Handle is perfect! Glad you got it. Let's ship this!